dependencies.yaml

```
@@ -696,7 +696,7 @@ dependencies:
       - output_types: [conda]
         packages:
           - *cugraph_unsuffixed
-          - pytorch>=2.0
+          - pytorch>=2.0,<2.4
```

```suggestion
          # ceiling could be removed when this is fixed:
          # https://github.com/conda-forge/pytorch-cpu-feedstock/issues/254
          - &pytorch_conda pytorch>=2.0,<2.4.0a0
```

Two suggestions:

1. could you please leave a comment in code here pointing back to the root cause, to hopefully make it easier for someone to notice that loosening this pin is worth investigating in the future?
2. To keep all these `pytorch` dependencies consistent, could you please use an anchor `&pytorch_conda` instead of `&pytorch_unsuffixed`? I know that use of `_unsuffixed` existed before this PR (probably my fault, honestly), but let's clean it up now that these exact lines are being touched.